### ISSUE TITLE
Support model dependent rule presets

### ISSUE THREAD

**Issue by MattHag** [pwr-Solaar/Solaar#2271]
**Information**

- Solaar version (`solaar --version` and `git describe --tags`): 1.1.10

**Is your feature request related to a problem? Please describe.**
The rule editor is not as straightforward as Logi Options+ to set up rules for button presses and gestures. Updates can improve that, but presets support. And presets for at least common models, such as the MX Master series would quickly help to get a basic configuration to play around with in the rule editor.

**Describe the solution you'd like**
Add presets support, which means

- Add some rules.yaml
- Somehow tie them to compatible mice e.g. MX Master, Master 2S, MX Master 3, MX Master 3S
- Ask to load presets, when a compatible device is connected the first time
  - Also support a way to switch them later. Somewhere related to the Rule editor button.

As an alternative or addition, It would also be possible to provide patterns, e.g. 

- I want to use the thumb wheel to switch tabs, provide me the correct trigger setup, so I only need to add the action to trigger.
- I want to use the gestures, provide me the rules for the Gesture button and gestures in the 4 directions, so I only need to add the action to execute.

**Describe alternatives you've considered**
Providing YAML files for download is also possible, but still cumbersome and not as simple as it should be.

**Additional context**
This doesn't avoid improvements to the rule editor, but should ensure that user do not drop Solaar due to the steep learning curve in setting up rules and grasping what the need for this and that functionality is.

**Comment by MattHag** [comment]
This ticket is one, that aims towards reducing the need for help requests like in #2326, which in the current state of Solaar 1.1.11 is totally understandable. The long term solution needs to be a device configuration, that does not require a manual to map key action etc. Everything that can automatically happen in the background needs to happen automatically.

**Comment by pfps** [comment]
What presets do you think would be universally helpful?  I'm having trouble coming up with any.

**Comment by MattHag** [comment]
Take the default configuration on macOS or Windows and it's a good start. On MX Master

- Gesture button -> Super_L
- Left/Right Gesture -> Switch desktop

But I'd prefer a more general approach to this, with just simplifying the interface, so it's user-friendly and doesn't have a steep learning curve.

**Comment by pfps** [comment]
Much of this is done in Linux by parts of the windowing system.   Solaar should not be in this business.

**Comment by MattHag** [comment]
That all maps to keyboards shortcuts, I am not sure if they differ. However, the current approach just takes too many clicks to setup some basic key mappings. Thus, the simplification of the rule editor should take a higher priority and then the need and feasibility of further improvements can be discussed.

**Comment by pfps** [comment]
Presets are only helpful if they are commonly used.   What are some good examples?

**Comment by MattHag** [comment]
Desktop switching on the left/right gesture buttons

**Comment by pfps** [comment]
I certainly don't want that.  And that's not tied to a particular device.

**Comment by MattHag** [comment]
Sure tied to a device otherwiese it makes no sense. But that's already implemented. You'll never find a config that's suitable, but the Logiops defaults are not bad and showcasing the same preconfigured options, which are easy to adopt is not bad. However the implementation needs to be well thought with good maintenence and testability in mind.

The UI should change to a per device like the main window anyways and special rules not appliccable can be in a General rules entry.

**Comment by pfps** [comment]
No, these rules work for any mouse that supports XY processing, not a particular mouse.  If I have multiple mice that support XY processing, I generally wouldn't want to have different rules for their mouse gestures.  Similarly, if I have multiple keyboards with a particular divertable key I wouldn't generally want to have different processing for this key on my different keyboards.  So I prefer having a global view of my rules.

**Comment by immanuelfodor** [comment]
About the defaults: I prefer CTRL+C and CTRL+V on the back/forward buttons :grinning:

**Comment by pfps** [comment]
It appears to me that there are very few rules that are universal and even among those few that are tied to a particular device model.   About the only examples I can think of are rules that enable the function keys that don't do anything but even there the enablement could easily be different for different distributions of Linux.

What would be useful is a library of rules that users could contribute to.   That could be part of the Solaar wiki.
